packages/types/src/data/abstract-query.ts

```diff
@@ -0,0 +1,240 @@
+	type: 'm2o';
+
+	join: AbstractQueryFieldNodeRelatedJoinMany;
+}
+
+export interface AbstractQueryFieldNodeRelatedOneToMany extends AbstractQueryNode, AbstractQueryFieldNodeRelatedBase {
+	type: 'o2m';
+
+	join: AbstractQueryFieldNodeRelatedJoinMany;
+}
+
+export interface AbstractQueryFieldNodeRelatedAnyToOne extends AbstractQueryNode, AbstractQueryFieldNodeRelatedBase {
+	type: 'a2o';
+
+	join: AbstractQueryFieldNodeRelatedJoinAny;
+}
+
+export interface AbstractQueryFieldNodeRelatedOneToAny extends AbstractQueryNode, AbstractQueryFieldNodeRelatedBase {
+	type: 'o2a';
+
+	join: AbstractQueryFieldNodeRelatedJoinAny;
+}
+
+// =================================================================================================
+
+/**
+ * Optional attributes to customize the query results
+ * @typeParam sort - Specifies the order of the results
+ * @typeParam limit - Specifies the maximum amount of returning results
+ */
+export interface AbstractQueryModifiers {
+	limit?: AbstractQueryNodeLimit;
+	offset?: AbstractQueryNodeOffset;
+	sort?: AbstractQueryNodeSort;
+	filter?: AbstractQueryNodeLogical | AbstractQueryNodeCondition;
+}
+
+/**
+ * Specifies the maximum amount of returning results
+ * @typeParam type - set to 'limit'
+ * @typeParam value - the limit value
+ */
+interface AbstractQueryNodeLimit {
+	type: 'limit';
+	value: number;
+}
+
+/**
+ * Specifies the number of items to skip before returning results
+ * @typeParam type - set to 'offset'
+ * @typeParam value - the offset value
+ */
+interface AbstractQueryNodeOffset {
+	type: 'offset';
+	value: number;
+}
+
+/**
+ * Specifies the order of the results
+ * @typeParam type - set to 'sort'
+ * @typeParam direction - 'ascending' or 'descending'
+ * @typeParam target - the node on which the sorting should be applied
+ */
+interface AbstractQueryNodeSort {
+	type: 'sort';
+	direction: 'ascending' | 'descending';
+	target:
+		| AbstractQueryFieldNodePrimitive
+		| AbstractQueryFieldNodeFn
+		| AbstractQueryFieldNodeRelatedManyToOne
+		| AbstractQueryFieldNodeRelatedAnyToOne;
+}
+
+/**
+ * Used to create logical operations
+ * @typeParam type - set to 'sort'
+ * @typeParam operator - the logical operator to use
+ * @typeParam children - left and right side of the operation.
+ * @example
+ * ```
+ * {
+ * 	type: 'logical',
+ * 	operator: 'and',
+ * 	children: [
+ * 		{
+ * 			type: 'condition',
+ * 			operation: 'eq',
+ * 			target: { type: 'field', key: 'a' }
+ * 			value: 5
+ * 		},
+ * 		{
+ * 			type: 'condition',
+ * 			operation: 'lt',
+ * 			target: { type: 'field', key: 'b' }
+ * 			value: 5
+ * 		}
+ *  ]
+ * }
+ * ```
+ */
+export interface AbstractQueryNodeLogical {
+	type: 'logical';
+	operator: 'and' | 'or' | 'not';
+	childNodes: (AbstractQueryNodeLogical | AbstractQueryNodeCondition)[];
+}
+
+/**
+ * Used to create conditional operations
+ * @typeParam type - set to 'condition'
+ * @typeParam target - the node on which the sorting should be applied
```

Comment:
```suggestion
 * @typeParam targetNode - the node on which the condition should be applied
```

(this also assumes we decide to use `targetNode`!)